x-pack/plugins/security_solution/public/resolver/data_access_layer/factory.ts

```diff
@@ -79,24 +79,39 @@ export function dataAccessLayerFactory(
       timeRange: TimeRange;
       indexPatterns: string[];
     }): Promise<ResolverPaginatedEvents> {
-      return context.services.http.post('/api/endpoint/resolver/events', {
-        query: { afterEvent: after, limit: 25 },
-        body: JSON.stringify({
-          timeRange: {
-            from: timeRange.from,
-            to: timeRange.to,
-          },
-          indexPatterns,
-          filter: JSON.stringify({
-            bool: {
-              filter: [
-                { term: { 'process.entity_id': entityID } },
-                { term: { 'event.category': category } },
-              ],
+      if (category === 'alerts') {
+        return context.services.http.post('/api/endpoint/resolver/events', {
```

Comment:
Thanks. Since these are calling the same api endpoint and have the same query config, can we just define the body in the if/else statement to de-dupe some of this logic?